src/index.ts

```diff
@@ -14,65 +14,70 @@ import { createConnection } from "typeorm";
   const { getErrorHandler } = require("login.dfe.express-error-handling");
   const routes = require("./routes");
 
-  createConnection().then(c => 
-    // run migrations on start up; only boot up if migrations succeed!
-    c.runMigrations({transaction: true})
-  ).then(() => {
-    const app = express();
-    app.use(
-      helmet({
-        noCache: true,
-        frameguard: {
-          action: "deny"
-        }
-      })
-    );
+  try {
+    const dbConnection = await createConnection();
+    // Run migrations on start up; only boot up if migrations succeed!
+    dbConnection.runMigrations({ transaction: true });
+  } catch (err) {
+    throw new Error(`Could not boot application, failed to connect to database, error: ${err}`);
+  }
 
-    if (config.hostingEnvironment.env !== "dev") {
-      app.set("trust proxy", 1);
-    }
+  const app = express();
+  app.use(
+    helmet({
+      noCache: true,
+      frameguard: {
+        action: "deny"
+      }
+    })
+  );
 
-    app.use(bodyParser.urlencoded({ extended: true }));
-    app.use(sanitization());
-    app.set("view engine", "ejs");
-    app.set("views", path.resolve(__dirname, "app"));
-    app.use(expressLayouts);
-    app.set("layout", "layouts/layout");
+  if (config.hostingEnvironment.env !== "dev") {
+    app.set("trust proxy", 1);
```

Comment:
should that line be in the else block at the bottom?

Reply:
I was thinking about the same thing yeah